Model/Order/CreditmemoFactory.php

```diff
@@ -0,0 +1,124 @@
+<?php
+/**
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the MIT License
+ * It is available through the world-wide-web at this URL:
+ * https://tldrlegal.com/license/mit-license
+ * If you are unable to obtain it through the world-wide-web, please send an email
+ * to support@buckaroo.nl so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade this module to newer
+ * versions in the future. If you wish to customize this module for your
+ * needs please contact support@buckaroo.nl for more information.
+ *
+ * @copyright Copyright (c) Buckaroo B.V.
+ * @license   https://tldrlegal.com/license/mit-license
+ */
+namespace Buckaroo\Magento2\Model\Order;
+
+use Magento\Bundle\Ui\DataProvider\Product\Listing\Collector\BundlePrice;
+use Magento\Sales\Api\Data\OrderItemInterface;
+use Magento\Sales\Model\Order\CreditmemoFactory as MagentoCreditmemoFactory;
+
+/**
+ * Factory class for @see \Magento\Sales\Model\Order\Creditmemo
+ */
+class CreditmemoFactory extends MagentoCreditmemoFactory
+{
+    /**
+     * Order convert object.
+     *
+     * @var \Magento\Sales\Model\Convert\Order
+     */
+    protected $convertor;
+
+    /**
+     * @var \Magento\Tax\Model\Config
+     */
+    protected $taxConfig;
+
+    /**
+     * @var \Magento\Framework\Unserialize\Unserialize
+     * @deprecated 101.0.0
+     */
+    protected $unserialize;
+
+    /**
+     * @var \Magento\Framework\Serialize\Serializer\Json
+     */
+    private $serializer;
+
+    protected $logger;
+
+    /**
+     * Factory constructor
+     *
+     * @param \Magento\Sales\Model\Convert\OrderFactory $convertOrderFactory
+     * @param \Magento\Tax\Model\Config $taxConfig
+     * @param \Magento\Framework\Serialize\Serializer\Json $serializer
+     */
+    public function __construct(
+        \Magento\Sales\Model\Convert\OrderFactory $convertOrderFactory,
+        \Magento\Tax\Model\Config $taxConfig,
+        \Magento\Framework\Serialize\Serializer\Json $serializer = null,
```

Comment:
please never put default parameters in constructors before obligatory ones

Reply:
Sure, thank you, fixed